lib/tapioca/dsl/compilers/active_record_fixtures.rb

```diff
@@ -104,12 +106,53 @@ def method_names_from_eager_fixture_loader
 
         sig { params(mod: RBI::Scope, name: String).void }
         def create_fixture_method(mod, name)
-          mod.create_method(
+          model_class = active_record_models[name] || "T.untyped"
+          mod.create_method_with_sigs(
             name,
-            parameters: [create_rest_param("fixture_names", type: "T.any(String, Symbol)")],
-            return_type: "T.untyped",
+            sigs: [
+              mod.create_sig(
+                parameters: {
+                  fixture_name: "T.any(String, Symbol)",
+                  other_fixtures: "NilClass",
+                },
+                return_type: model_class,
+              ),
+              mod.create_sig(
+                parameters: {
+                  fixture_name: "T.any(String, Symbol)",
+                  other_fixtures: "T.any(String, Symbol)",
+                },
+                return_type: "T::Array[#{model_class}]",
+              ),
+            ],
+            parameters: [
+              RBI::ReqParam.new("fixture_name"),
+              RBI::RestParam.new("other_fixtures"),
+            ],
           )
         end
+
+        sig { returns(T::Hash[String, String]) }
+        def active_record_models
+          @active_record_models ||= T.let(
+            ActiveRecord::Base.descendants.map do |model|
+              [fixture_name(model.name), model.name]
+            end.to_h,
+            T.nilable(T::Hash[String, T::Class[ActiveRecord::Base]]),
```

Comment:
Is this type correct? It looks like we store the `model.name` right?

```suggestion
            T.nilable(T::Hash[String, String]),
```

Reply:
yeah, at first it was a mapping to the classes, I am also going to check why it did not complain about the types because it really should have